src/precompiles/ecrecover.rs

```diff
@@ -0,0 +1,230 @@
+use zkevm_opcode_defs::k256::ecdsa::VerifyingKey;
+pub use zkevm_opcode_defs::sha2::Digest;
+use zkevm_opcode_defs::{ethereum_types::U256, k256, sha3};
+
+use crate::eravm_error::EraVmError;
+
+use super::*;
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ECRecoverPrecompile;
+
+impl Precompile for ECRecoverPrecompile {
+    fn execute_precompile(&mut self, query: U256, heaps: &mut Heaps) -> Result<(), EraVmError> {
+        let precompile_call_params = query;
+        let params = precompile_abi_in_log(precompile_call_params);
+
+        let mut current_read_location = MemoryLocation {
+            page: params.memory_page_to_read,
+            index: params.output_memory_offset,
+        };
+
+        let hash_query = MemoryQuery {
+            location: current_read_location,
+            value: U256::zero(),
```

Comment:
Do we know why this MemoryQuery needs a zero?

Reply:
It is the default value, it gets replaced when executing the queries.

Reply:
Super.